hypothesis-python/src/hypothesis/database.py

```diff
@@ -225,17 +225,20 @@ def save(self, key: bytes, value: bytes) -> None:
         # Note: we attempt to create the dir in question now. We
         # already checked for permissions, but there can still be other issues,
         # e.g. the disk is full, or permissions might have been changed.
-        self._key_path(key).mkdir(exist_ok=True, parents=True)
-        path = self._value_path(key, value)
-        if not path.exists():
-            suffix = binascii.hexlify(os.urandom(16)).decode("ascii")
-            tmpname = path.with_suffix(f"{path.suffix}.{suffix}")
-            tmpname.write_bytes(value)
-            try:
-                tmpname.rename(path)
-            except OSError:  # pragma: no cover
-                tmpname.unlink()
-            assert not tmpname.exists()
+        try:
+            self._key_path(key).mkdir(exist_ok=True, parents=True)
+            path = self._value_path(key, value)
+            if not path.exists():
+                suffix = binascii.hexlify(os.urandom(16)).decode("ascii")
+                tmpname = path.with_suffix(f"{path.suffix}.{suffix}")
+                tmpname.write_bytes(value)
```

Comment:
I saw this `write_bytes` fail occasionally with "file not found" in a pandas runner, here and in other PRs.

I don't know why it happens, or why it seems to happen mostly with pandas, but it looks like silently ignoring IO failures here is in the spirit of things.